Add '-pn' option to specify port name.

mckuhei opened the pull request:
E.g `sc55emu.exe "-pn:2- Roland SC-D70 MIDI IN"` to open midi devices, If devices not exist, will fallback to `-p` option.

mckuhei committed:
Add '-pn' option to specify port name.

NicknineTheEagle commented on src/mcu.cpp:
```diff
@@ -1354,6 +1358,28 @@ int main(int argc, char *argv[])
+        auto end = devices + length;
+        while (start != end) {
+            int len = strlen(start);
+            if (!strcmp(start, pname)) {
```
IMHO, it's better to use case insensitive check here (_stricmp on MSVC, strcasecmp on POSIX).

mckuhei replied:
ok

mckuhei commented:
ok

mckuhei committed:
Use case insensitive string comparison.

mckuhei commented:
@nukeykt Please close or merge this pull request.

mckuhei commented:
It's very annoying to use port number open a midi devices, especially if you have a hardware device.

UnBeatWaterGH commented:
> command line arguments probably will be redundant once gui and config files are added, these are being done and will be in next version.

I don't think command line arguments will be _completely_ useless.

mckuhei committed:
Merge branch 'nukeykt:master' into master

mckuhei committed:
Merge branch 'nukeykt:master' into master

mckuhei committed:
Add test background for SC-55mkII

mckuhei committed:
Fix unable select port by name in release mode

mckuhei committed:
Add ALL MUTE Led

mckuhei committed:
Forgot to upload image

mckuhei committed:
Add STANDBY light and Volume knob

mckuhei committed:
Add working volume knob

mckuhei committed:
Add LCD Contrast

mckuhei committed:
Improve LCD contrast

mckuhei committed:
I forgot upload background again

mckuhei committed:
Make LCD contrast simulation more accurate.

mckuhei committed:
Add clickable button

mckuhei committed:
Add MIDI output

mckuhei committed:
Fix font overlay

mckuhei committed:
Add serial port support.

mckuhei committed:
Improve serial port support.

linoshkmalayil committed:
Added serial IO support for POSIX based systems

mckuhei committed:
Merge pull request #2 from linoshkmalayil/linosh_linux_serial_support

Add Linux Serial Support

mckuhei committed:
Update Gui code

Rework volume knob.
Add midi lock.
Add LCD pixel blur.
Save sram state on exit.

mckuhei committed:
Change initialize memory behavior

Also change submcu timer
